Feature/segasaway/module9

gasawase opened the pull request:
**Module / Assignment**

Module 9: Create Lobby Logic and Chat solution

**Description**

Created social functionality of kicking, readying up, and leaving a lobby. Also managing of hidding/displaying buttons based on the status of all of the players. When all players are readied up, the start button appears on the host's screen and clicking it will spawn the players.

**Tech Notes**

Still have not implemented ownership of the camera changing; will implement next PR.

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

**Screenshot if applicable**

![Movie_007](https://github.com/user-attachments/assets/ff6d584c-4d2b-42c2-95a4-e85892685516)


**Checklist**
- [X] My code follows the style guidelines of this project
- [ ] My changes generate no new warnings (Generates a warning because I removed a registered prefab from the nw vars list)
- [X] Develop has been merged into this branch
- [X] Project builds and runs in Unity Editor
- [X] Project builds and runs on Windows Standalone

gasawase committed:
module 9 - part 1

- fixed animation timings
- fixed messy physics and controls (removed character controller; instead movement is determined by the rigidbody)

gasawase committed:
Module 9 Player Lobby

- manage enable/disable social buttons based on ready and added status of players
- created kicking player function
- created ready up function
- add player panels to group of players based on whether they are ready or not
- manage button visibility based on status of all players
- start game triggers spawning of players

MRN-DX reviewed (approved):
-Kicking, leaving, player lobby syncs with ready up. 
-All players ready logic working
-Chat solution is missing which is a big component to this module
-55/70, can turn in another update commit from this PR for the other 15 points for the chat solution

gasawase committed:
Added Chat Capabilities

- added text box to the content in the chat panel
- added new method to a listener on the send button
- send an rpc call on send, erase the input field on send, append the text to the chat panel on send
- rpc calls are sent to everyone

gasawase commented:
I added the chat capabilities now (I completely missed the last few minutes of the video). Let me know if I need to change anything else!

MRN-DX commented:
I will adjust grade, I was able to get the working chat functionality tested.
-Simple solution using ClientID to be the name, which is acceptable.
-Some improvements could be isolating out the chat solution in its own script in case you wanted to reuse the chat in other scenarios or moments in the game.
-If we wanted a solution that allowed players to change their name we would have to add additional methods to the Base PlayerData, set the name, and then access it through the networkedData, which would have been the smarter route of streamline any user data to one source.
15+ points
